docs/updater.md

```diff
@@ -0,0 +1,102 @@
+
+### Examples
+
+```
+mPipeline.enable_shared_ownership(); // Make it usable with the updater
+mUpdater->on(gvk::swapchain_resized_event(gvk::context().main_window()), 
+      gvk::shader_files_changed_event(mPipeline))
+  .update(mPipeline);
+```
+
+In the above example, the `gvk::updater mUpdater` is configured so that the `mPipeline` is updated when at least one of two _events_ occurs: either when the swapchain of the main window has been resized, or when one of the `mPipeline`'s shader files have been changed on the file system. 
+
+```
+mUpdater->on(
+      gvk::swapchain_format_changed_event(gvk::context().main_window()),
+      gvk::swapchain_additional_attachments_changed_event(gvk::context().main_window())
+   ).invoke([this]() {
+      // update render pass here
+   }).then_on(
+      gvk::swapchain_changed_event(gvk::context().main_window()),
+      gvk::shader_files_changed_event(mPipeline)
+   ).update(mPipeline);
+```
+
+In the above example, the renderpass belonging to `mPipeline` requires recreation only if the events `swapchain_format_changed_event` or `swapchain_format_changed_event` arise. Then it is evaluated whether _any type_ of swapchain recreation has occured and if so, subsequently the pipeline is updated. It is important to note here: The evaluation of events given to `then_on()` is only temporaly restricted by the previous list: this evaluation will occur in any case.
+
+## Common Use-Cases
+* Swapchain recreation can lead to changes to image properties. This may cause any pipeline, renderpass, framebuffer or image view that directly works with swapchain to end in an invalid state. The updater mechanism is meant to streamline the recreation process of those objects, as well as to allow any user-defined behaviour to execute in cases where such events occur.
+* Shader hot reloading: Update graphics/compute/ray-tracing pipelines after (SPIR-V) shader files have been updated on the file system.
+* Adapt to number of concurrent frame changes: setup any necessary sync mechanism.
+
+## Notes on Swapchain recreation
+
+* Modifying the following attributes may trigger a swapchain recreation:
+  * window size,
+  * presentation mode,
+  * number of presentation images,
+  * number of concurrent frames,
+  * backbuffer image format,
+  * out of data, or suboptimal swapchain detection by vulkan.
+* The swapchain can only resize if the `window` has been configured to allow being resized (via `window::enable_resizing`).   
+
+## General Notes
+* The updater mechanism is invoked just before the `render()` call of the associated `gvk::invokee`.
+* When you hand over resources that shall be automatically updated to the `gvk::updater`, the `gvk::updater` needs to take ownership of them. Therefore, you'll have to pay the price of enabling shared resource ownership (by the means of `avk::owning_resource::enable_shared_ownership`) for all of these resources.
```

Comment:
```suggestion
* When resources are handed over to the updater mechanism to be updated automatically, the `gvk::updater` needs to take ownership of them. Therefore, you'll have to pay the price of enabling shared resource ownership (by the means of `avk::owning_resource::enable_shared_ownership`) for all of these resources.
```

It should be noted that the convenience function `avk::shared` can be used to pass resources for that purpose.